controllers/addon_controller.go

```diff
@@ -359,13 +360,25 @@ func ignoreNotFound(err error) error {
 	return err
 }
 
+// runWorkflow returns addonmgrv1alpha1.Succeeded only if template is empty or returns corresponding lifecycle status if already completed
 func (r *AddonReconciler) runWorkflow(ctx context.Context, lifecycleStep addonmgrv1alpha1.LifecycleStep, addon *addonmgrv1alpha1.Addon, wfl workflows.AddonLifecycle) (addonmgrv1alpha1.ApplicationAssemblyPhase, error) {
 	log := r.Log.WithValues("addon", fmt.Sprintf("%s/%s", addon.Namespace, addon.Name))
 
+	if lifecycleStep == addonmgrv1alpha1.Prereqs && addon.GetPrereqStatus().Completed() {
+		log.Info("Lifecycle completed, skipping workflow execution", "lifecycleStep", lifecycleStep)
+		return addon.GetPrereqStatus(), nil
+	} else if lifecycleStep == addonmgrv1alpha1.Install && addon.GetInstallStatus().Completed() {
+		log.Info("Lifecycle completed, skipping workflow execution", "lifecycleStep", lifecycleStep)
+		return addon.GetInstallStatus(), nil
+	} else if lifecycleStep == addonmgrv1alpha1.Delete && (addon.Status.Lifecycle.Installed == addonmgrv1alpha1.DeleteFailed || addon.Status.Lifecycle.Installed == addonmgrv1alpha1.DeleteSucceeded) {
```

Comment:
```suggestion
	} else addon.Deleting() {
```
You might be able to just handle Deleting states more simply.

Reply:
actually `runWorkflow` wants to generate wf obj if install status is simply `Deleting`; so will need to explicitly check for `DeleteSucceeded` or `DeleteFailed`